### DISCUSSION TITLE
For production Image Optimization with Next.js, the optional 'sharp' package is strongly recommended.

### DISCUSSION THREAD

**Discussion by vwalexanderjin** [vercel/next.js#66303] [Help]
### Summary

Hey guys ! I think I need some help.

According to [Nextjs official docs](https://nextjs.org/docs/messages/sharp-missing-in-production),  Already installed sharp, but still have this in pm2 logs when in production mode.

Error Logs Details:
![1](https://github.com/vercel/next.js/assets/12606338/de12ff84-22b2-4e28-9ae2-8c3457640d96)

### Additional information

```js
Environment:
next: "^14.2.3",
sharp: "^0.33.4"

front-end dev:
nextjs + pm2 + node.js + docker
```

**Comment by icyJoseph** [comment]
The docs also talk about how, sharp can sometimes, not be resolved. You can help that by providing a path to the node_modules, like, `export NEXT_SHARP_PATH=/tmp/node_modules/sharp` in this case the `/tmp` is actually dependent on your deploy environment.

**Reply by snappytux** [reply]
same problem

## Dockerfile

```dockerfile
FROM --platform=linux/amd64 node:20-alpine
ENV PORT 3000
ENV NEXT_SHARP_PATH=/usr/app/node_modules/sharp
RUN apk add --update --no-cache tzdata curl && \
  cp /usr/share/zoneinfo/Asia/Bangkok /etc/localtime && \
  apk del tzdata && \
  addgroup -S nonroot && \
  adduser -S nonroot -G nonroot && \
  rm -rf /var/cache/apk/* /etc/apk/cache/*
WORKDIR /usr/app
COPY --chown=nonroot:nonroot ./public ./public
COPY --chown=nonroot:nonroot ./.next/standalone ./
COPY --chown=nonroot:nonroot ./.next/static ./.next/static
RUN chmod -R g+X /usr/app/
USER nonroot
HEALTHCHECK --interval=60s --timeout=3s --start-period=5s \
  CMD curl --fail http://localhost:3000 || exit 1
EXPOSE 3000
CMD HOSTNAME="0.0.0.0" node server.js
```

## docker run

<img width="990" alt="Screen Shot 2567-05-30 at 15 13 44" src="https://github.com/vercel/next.js/assets/7285460/de43914c-d78e-4b9b-af56-5e2c1ca3709b">

## Additional information
```
Environment:
next: "^14.2.3",
sharp: "^0.33.4"
```

**Comment by cluke009** [comment]
I recently tried upgrading my sharp install to ^0.33 and the build complained for me as well. I ended up just downgrading back ^0.32 and everything built properly again. Might be worth a shot for you.

**Reply by alsherif-khalaf** [reply]
> I recently tried upgrading my sharp install to ^0.33 and the build complained for me as well. I ended up just downgrading back ^0.32 and everything built properly again. Might be worth a shot for you.

worked for me too
